subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -244,20 +245,28 @@ int ec_host_cmd_init(struct ec_host_cmd_backend *backend)
 
 	handler_tx_buf = hc->tx.buf;
 	handler_rx_buf = hc->rx_ctx.buf;
+	tx_buf_end = handler_tx_buf + CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_SIZE;
+	rx_buf_end = handler_rx_buf + CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_SIZE;
 
 	ret = backend->api->init(backend, &hc->rx_ctx, &hc->tx);
 
 	if (ret != 0) {
 		return ret;
 	}
 
-	if (!hc->tx.buf | !hc->rx_ctx.buf) {
+	if (!hc->tx.buf || !hc->rx_ctx.buf) {
 		LOG_ERR("No buffer for Host Command communication");
 		return -EIO;
 	}
 
-	if ((handler_tx_buf && (handler_tx_buf != hc->tx.buf)) ||
-	    (handler_rx_buf && (handler_rx_buf != hc->rx_ctx.buf))) {
+	/* Check if a backend uses provided buffers. The buffer pointers can be shifted within the
+	 * buffer to make space for preamble. Make sure the rx/tx pointers are within the provided
+	 * buffers ranges.
+	 */
+	if ((handler_tx_buf && !((handler_tx_buf <= (uint8_t *)hc->tx.buf) &&
+				(tx_buf_end > (uint8_t *)hc->tx.buf))) ||
+	    (handler_rx_buf && !((handler_rx_buf <= hc->rx_ctx.buf) &&
+				(rx_buf_end < hc->rx_ctx.buf)))) {
```

Comment:
Would it make sense to have another pair of local variables for `hc->tx.buf` and `hc->rx_ctx.buf`?  Something descriptive, just trying to make the condition easier to understand, it's a bit puzzling that first you set them to `handler_tx_buf` but then they change from `init()`... maybe `backend_tx_buf` and `backend_rx_buf`?

Reply:
Make sense if you think it is easier to understand.